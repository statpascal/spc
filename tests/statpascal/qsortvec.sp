program qsortvec;

type 
    element = int32;
    vec = vector of element;

function qsort (a: vec): vec;
    begin
        if size (a) <= 1 then
            qsort := a
        else
           qsort := combine (
               qsort (a [a < a [1]]), 
               a [a = a [1]], 
               qsort (a [a > a [1]]))
    end;

const
    n = 20;
var 
    a: vec;

begin
    a := randomperm (n);
    writeln ('Unsorted: ', a);
    writeln ('Sorted:   ', qsort (a))
end.
