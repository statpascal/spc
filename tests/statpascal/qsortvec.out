Unsorted: 10 7 12 6 9 11 2 14 5 1 4 8 16 13 18 19 15 20 3 17 
Sorted:   1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 
